src/components/ItemList.svelte

```diff
@@ -28,6 +29,16 @@
   let topElement;
   let scrollY;
 
+  function fullTextSearch(query) {
+    // create an index of item names and descriptions for full-text search
+    const searchIndex = new Index();
+    items.forEach((item) => searchIndex.add(item.name, item.description));
```

Comment:
I think we should generate the index as early as possible. Re-generating it everytime there's a a search is unnecessary and might be slow.